target/arm/helper.c

```diff
@@ -8189,6 +8866,184 @@ void register_cp_regs_for_features(ARMCPU *cpu)
         define_arm_cp_regs(cpu, ccsidr2_reginfo);
     }
 
+#ifdef TARGET_CHERI
+    // Claim is not really CHERI specific
+    define_arm_cp_regs(cpu, claim_cp_reginfo);
+
+    // LETODO: Stuff to do with CPACR_ELX.CEN / EN for stopping DDC access, also
+    // HCR controls a lot of these LETODO: Also have to pay attention to
+    // restricted for RDDC and RSP.
+    cap_register_t null_cap;
+    bzero(&null_cap, sizeof(null_cap));
+    cap_register_t max_cap = cc128_make_max_perms_cap(0, 0, CAP_MAX_TOP);
+    ARMCPRegInfo cheri_regs[] = {
+        // We swap the relevent DDC into this register, so it is always
+        // accessible
+        {.name = "DDC",
```

Comment:
Not sure if there is a clang-format option to do that, but I think we can deal with it when we merge from upstream. Reverting the formatting while keeping the changes is really tedious.

Reply:
This one I care about less because it's never going to conflict, though should still be done in the same style as the others. I do care about the other big tables, it makes it impossible to read the diff.